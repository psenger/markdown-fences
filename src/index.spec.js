const {join} = require('path');
const tmp = require('tmp');

const {
  readContents,
  writeContents,
  injectJsDoc,
  injectFileFencePosts,
  injectToc,
  generateMarkDownFile
} = require('./index');

const cp = ( src, dest ) => require("fs").copyFileSync(src, dest);
const rm = (src) => require('fs').unlinkSync(src);

describe('@pseger/markdown-fences', () => {
  describe('#readContents', () => {
    it('should fail when the required parameter is undefined', async () => {
      expect.assertions(1);
      return expect(()=>readContents()).rejects.toThrow();
    })
    it('should fail when the required parameter is null', async () => {
      expect.assertions(1);
      return expect(()=>readContents(null)).rejects.toThrow();
    })
    it('should fail when the required parameter is missing', async () => {
      expect.assertions(1);
      return expect(()=>readContents('booboo.mac.goo')).rejects.toThrow();
    })
    it('should read the file sample.md from the test harness directory', async () => {
      expect.assertions(1);
      const results = await readContents(join(__dirname,'__test_harness__','sample.md'));
      return expect(results).toMatchSnapshot();
    })
  })
  describe('#writeContents', () => {
    it('should fail when the required parameter is undefined', async () => {
      expect.assertions(1);
      return expect(()=>writeContents()).rejects.toThrow();
    })
    it('should fail when the required parameter is null', async () => {
      expect.assertions(1);
      return expect(()=>writeContents(null, null)).rejects.toThrow();
    })
    it('should just work reading and writting', async () => {
      expect.assertions(1);
      const tmpobj = tmp.fileSync();
      const file = tmpobj.name;
      await writeContents( file, 'hello')
      const results = await readContents(file);
      expect(results).toMatchInlineSnapshot('"hello"');
      tmpobj.removeCallback();
    })
  })
  describe('#injectJsDoc', ()=>{
    it('should skip the block',async ()=>{
      expect.assertions(1);
      const readMe = `# Start

<!--START_SECTION:deep1/deep2/insert.md-->
<!--END_SECTION:deep1/deep2/insert.md-->

## End`
      let results =  await injectJsDoc(readMe, [ join(__dirname,'__test_harness__','foo.js') ])
      expect(results).toMatchSnapshot();
    })
    it('replace the tags',async ()=>{
      expect.assertions(1);
      const readMe = `# Start

<!--START_SECTION:jsdoc-->
<!--END_SECTION:jsdoc-->

## End`
      let results =  await injectJsDoc(readMe, [ join(__dirname,'__test_harness__','foo.js') ])
      expect(results).toMatchSnapshot();
    })
  })
  describe('injectToc',() => {
    it('should skip the block',async ()=>{
      expect.assertions(1);
      const readMe = `# Start

<!--START_SECTION:blaablaa-->
<!--END_SECTION:blaablaa-->

## Level 2-A

### Level 3-A

### Level 3-B

### Level 3-C

## Level 2-B

### Level 3-A

#### Level 4-A

#### Level 4-B

### Level 3-C

## Level 2-C

`
      let results =  await injectToc(readMe)
      expect(results).toMatchSnapshot();
    })
    it('replace the tags',async ()=>{
      expect.assertions(1);
      const readMe = `# Start

<!--START_SECTION:toc-->
<!--END_SECTION:toc-->

## Level 2-A

### Level 3-A

### Level 3-B

### Level 3-C

## Level 2-B

### Level 3-A

#### Level 4-A

#### Level 4-B

### Level 3-C

## Level 2-C

`
      let results =  await injectToc(readMe)
      expect(results).toMatchSnapshot();
    })
  });
  describe('#injectFileFencePosts',()=>{
    it('Testing full directory',async () => {
      const buildDynamicReadMe = (file) => `# Start

<!--START_SECTION:file:${file}-->
<!--END_SECTION:file:${file}-->

## End
`
      const tmpobj = tmp.dirSync();
      const tempDirectoryAndFileLocation = join( tmpobj.name, 'sample.md');
      cp( join( __dirname,'__test_harness__','sample.md'), tempDirectoryAndFileLocation )
      const results = await injectFileFencePosts(buildDynamicReadMe('sample.md'), tmpobj.name, {log:false})
      expect(results).toMatchSnapshot();
      rm( tempDirectoryAndFileLocation )
      tmpobj.removeCallback();
    })
    it('Testing failed directory',async () => {
      const buildDynamicReadMe = (file) => `# Start

<!--START_SECTION:file:${file}-->
<!--END_SECTION:file:${file}-->

## End
`
      const tmpobj = tmp.dirSync();
      const results = await injectFileFencePosts(buildDynamicReadMe('sample.md'), tmpobj.name, {log:true})
      expect(results).toMatchSnapshot();
      tmpobj.removeCallback();
    })
  })
  describe('generateMarkDownFile',()=>{
    it('full integration test',async ()=>{
      const tmpobj = tmp.dirSync();
      await generateMarkDownFile(
        join( __dirname, '__test_harness__', 'generateMarkDownFile', '.README.md' ),
        join( tmpobj.name, 'README.md' ),
        join( __dirname, '__test_harness__', 'generateMarkDownFile' ),
        [ join( __dirname, '__test_harness__', 'foo.js') ]
      )
      const results = await readContents( join( tmpobj.name, 'README.md' ) );
      expect(results).toMatchSnapshot();
      rm( join( tmpobj.name, 'README.md' ) )
      tmpobj.removeCallback();
    })
  })
})
