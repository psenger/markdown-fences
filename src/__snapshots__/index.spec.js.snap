// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@pseger/markdown-fences #injectCodeFencePosts Testing failed directory with language called out 1`] = `
"# Start

<!--START_CODE_FENCE_SECTION:javascript:file:/foo.js-->
<!--END_CODE_FENCE_SECTION:javascript:file:/foo.js-->

## End
"
`;

exports[`@pseger/markdown-fences #injectCodeFencePosts Testing failed directory without language called out 1`] = `
"# Start

<!--START_CODE_FENCE_SECTION:file:/foo.js-->
<!--END_CODE_FENCE_SECTION:file:/foo.js-->

## End
"
`;

exports[`@pseger/markdown-fences #injectCodeFencePosts Testing full directory with language called out 1`] = `
"# Start

<!--START_CODE_FENCE_SECTION:javascript:file:/foo.js-->
\`\`\`javascript
/**
 * @typedef boom
 * @property {String} namnam - nom nom
 * @property {String[]} keys - the keys
 */

/**
 * Foo test
 * @type {{boo: exports.boo, foo: exports.foo}}
 */
module.exports = {
  /**
   * foo test
   * @param {String} st - something
   * @param {boolean} end - more of something
   * @return {Promise<boolean>}
   */
  foo: async function (st, end) {
    return Promise.resolve(true)
  },
  /**
   * boo test
   * @param {String} [st] - something
   * @param {boolean} [end=false] - more something
   * @return {boom}
   */
  boo: function (st, end) {
    return { namnam: 'xxx', keys: ['a'] }
  }
}

\`\`\`
<!--END_CODE_FENCE_SECTION:javascript:file:/foo.js-->

## End
"
`;

exports[`@pseger/markdown-fences #injectCodeFencePosts Testing full directory without language called out 1`] = `
"# Start

<!--START_CODE_FENCE_SECTION:file:/foo.js-->
\`\`\`
/**
 * @typedef boom
 * @property {String} namnam - nom nom
 * @property {String[]} keys - the keys
 */

/**
 * Foo test
 * @type {{boo: exports.boo, foo: exports.foo}}
 */
module.exports = {
  /**
   * foo test
   * @param {String} st - something
   * @param {boolean} end - more of something
   * @return {Promise<boolean>}
   */
  foo: async function (st, end) {
    return Promise.resolve(true)
  },
  /**
   * boo test
   * @param {String} [st] - something
   * @param {boolean} [end=false] - more something
   * @return {boom}
   */
  boo: function (st, end) {
    return { namnam: 'xxx', keys: ['a'] }
  }
}

\`\`\`
<!--END_CODE_FENCE_SECTION:file:/foo.js-->

## End
"
`;

exports[`@pseger/markdown-fences #injectFileFencePosts Testing failed directory 1`] = `
"# Start

<!--START_SECTION:file:sample.md-->
<!--END_SECTION:file:sample.md-->

## End
"
`;

exports[`@pseger/markdown-fences #injectFileFencePosts Testing full directory 1`] = `
"# Start

<!--START_SECTION:file:sample.md-->
# Sample File

## First Heading

Lorem ipsum dolor sit amet, consectetur adipiscing elit. Pellentesque metus massa, ultrices nec nulla in, commodo feugiat quam. Etiam semper semper nisi at ultrices. Vestibulum blandit enim vitae ex consectetur, eu dignissim lorem vulputate. In sit amet dictum metus, in sollicitudin mi. Phasellus finibus velit sapien, at fringilla odio vehicula eget. Proin venenatis bibendum dictum. Ut iaculis nulla ac neque molestie, eget semper lorem molestie. Maecenas interdum arcu eu diam vehicula elementum. Etiam iaculis erat at nulla fringilla, non ullamcorper libero lobortis. Suspendisse potenti. Ut vitae elementum diam. Aliquam nec mauris iaculis, placerat ligula ut, dapibus justo. Nulla sodales urna vel est eleifend interdum. Sed vulputate, justo sit amet facilisis tristique, dolor tellus tincidunt nibh, sit amet aliquet turpis risus quis est.

Sed ut ante malesuada, varius velit dignissim, suscipit neque. Nullam id nulla a velit iaculis lobortis. Nam finibus diam libero, eget blandit augue lacinia at. Etiam cursus ut nibh vel dignissim. Praesent consectetur in enim eu vestibulum. Ut vestibulum pellentesque quam non pulvinar. Mauris hendrerit molestie lacinia. Sed elementum tortor suscipit, pharetra nisl at, pellentesque purus.

Fusce sagittis sem sit amet nunc condimentum pharetra. Cras in enim commodo, pretium neque a, scelerisque elit. Etiam et feugiat lorem. Maecenas tempus, eros vel cursus commodo, neque erat placerat nisl, nec scelerisque ipsum urna ac ante. In purus odio, feugiat quis vestibulum at, tempor nec neque. Quisque laoreet sapien in ex malesuada porttitor eu ut nisl. Etiam hendrerit elementum bibendum. Aliquam pharetra feugiat porttitor. Aliquam convallis dolor eu porta pellentesque. Proin ultricies tempor ex, ac convallis est ornare et. Donec sodales, ex nec sodales rhoncus, nunc metus facilisis massa, sit amet dapibus eros neque eget dolor. Nunc dignissim nulla ex, et luctus ante semper sit amet. Aenean id felis porta, facilisis arcu id, malesuada ex. Aenean scelerisque magna vitae posuere consectetur.

Curabitur at euismod nunc. Praesent sagittis lorem quis rutrum ornare. Vivamus ut viverra ligula. Aenean porta, lectus non pretium aliquam, leo ante varius quam, a sodales lectus urna ut tellus. In id nulla libero. Maecenas ut tempor justo. Integer diam erat, consectetur tempor dictum sed, lacinia ac elit. Vestibulum porta nulla odio, in vestibulum quam volutpat quis.

Nunc pulvinar odio in sapien molestie, quis elementum dolor porta. Nunc dapibus rutrum eros, ac varius elit rutrum ac. Cras et facilisis nunc, et consequat risus. Vivamus id maximus quam, et lacinia magna. In aliquet auctor sem. Duis finibus nisi ligula, vel tempor magna vehicula at. In neque tellus, ornare quis augue sit amet, luctus malesuada metus.

## Second Heading

<!--START_SECTION:jsdoc-->
<!--END_SECTION:jsdoc-->

## Third Heading

Lorem ipsum dolor sit amet, consectetur adipiscing elit. Pellentesque metus massa, ultrices nec nulla in, commodo feugiat quam. Etiam semper semper nisi at ultrices. Vestibulum blandit enim vitae ex consectetur, eu dignissim lorem vulputate. In sit amet dictum metus, in sollicitudin mi. Phasellus finibus velit sapien, at fringilla odio vehicula eget. Proin venenatis bibendum dictum. Ut iaculis nulla ac neque molestie, eget semper lorem molestie. Maecenas interdum arcu eu diam vehicula elementum. Etiam iaculis erat at nulla fringilla, non ullamcorper libero lobortis. Suspendisse potenti. Ut vitae elementum diam. Aliquam nec mauris iaculis, placerat ligula ut, dapibus justo. Nulla sodales urna vel est eleifend interdum. Sed vulputate, justo sit amet facilisis tristique, dolor tellus tincidunt nibh, sit amet aliquet turpis risus quis est.

Sed ut ante malesuada, varius velit dignissim, suscipit neque. Nullam id nulla a velit iaculis lobortis. Nam finibus diam libero, eget blandit augue lacinia at. Etiam cursus ut nibh vel dignissim. Praesent consectetur in enim eu vestibulum. Ut vestibulum pellentesque quam non pulvinar. Mauris hendrerit molestie lacinia. Sed elementum tortor suscipit, pharetra nisl at, pellentesque purus.

Fusce sagittis sem sit amet nunc condimentum pharetra. Cras in enim commodo, pretium neque a, scelerisque elit. Etiam et feugiat lorem. Maecenas tempus, eros vel cursus commodo, neque erat placerat nisl, nec scelerisque ipsum urna ac ante. In purus odio, feugiat quis vestibulum at, tempor nec neque. Quisque laoreet sapien in ex malesuada porttitor eu ut nisl. Etiam hendrerit elementum bibendum. Aliquam pharetra feugiat porttitor. Aliquam convallis dolor eu porta pellentesque. Proin ultricies tempor ex, ac convallis est ornare et. Donec sodales, ex nec sodales rhoncus, nunc metus facilisis massa, sit amet dapibus eros neque eget dolor. Nunc dignissim nulla ex, et luctus ante semper sit amet. Aenean id felis porta, facilisis arcu id, malesuada ex. Aenean scelerisque magna vitae posuere consectetur.

Curabitur at euismod nunc. Praesent sagittis lorem quis rutrum ornare. Vivamus ut viverra ligula. Aenean porta, lectus non pretium aliquam, leo ante varius quam, a sodales lectus urna ut tellus. In id nulla libero. Maecenas ut tempor justo. Integer diam erat, consectetur tempor dictum sed, lacinia ac elit. Vestibulum porta nulla odio, in vestibulum quam volutpat quis.

Nunc pulvinar odio in sapien molestie, quis elementum dolor porta. Nunc dapibus rutrum eros, ac varius elit rutrum ac. Cras et facilisis nunc, et consequat risus. Vivamus id maximus quam, et lacinia magna. In aliquet auctor sem. Duis finibus nisi ligula, vel tempor magna vehicula at. In neque tellus, ornare quis augue sit amet, luctus malesuada metus.


## Fourth Heading

<!--START_SECTION:deep1/deep2/insert.md-->
<!--END_SECTION:deep1/deep2/insert.md-->

## Third Heading

Lorem ipsum dolor sit amet, consectetur adipiscing elit. Pellentesque metus massa, ultrices nec nulla in, commodo feugiat quam. Etiam semper semper nisi at ultrices. Vestibulum blandit enim vitae ex consectetur, eu dignissim lorem vulputate. In sit amet dictum metus, in sollicitudin mi. Phasellus finibus velit sapien, at fringilla odio vehicula eget. Proin venenatis bibendum dictum. Ut iaculis nulla ac neque molestie, eget semper lorem molestie. Maecenas interdum arcu eu diam vehicula elementum. Etiam iaculis erat at nulla fringilla, non ullamcorper libero lobortis. Suspendisse potenti. Ut vitae elementum diam. Aliquam nec mauris iaculis, placerat ligula ut, dapibus justo. Nulla sodales urna vel est eleifend interdum. Sed vulputate, justo sit amet facilisis tristique, dolor tellus tincidunt nibh, sit amet aliquet turpis risus quis est.

Sed ut ante malesuada, varius velit dignissim, suscipit neque. Nullam id nulla a velit iaculis lobortis. Nam finibus diam libero, eget blandit augue lacinia at. Etiam cursus ut nibh vel dignissim. Praesent consectetur in enim eu vestibulum. Ut vestibulum pellentesque quam non pulvinar. Mauris hendrerit molestie lacinia. Sed elementum tortor suscipit, pharetra nisl at, pellentesque purus.

Fusce sagittis sem sit amet nunc condimentum pharetra. Cras in enim commodo, pretium neque a, scelerisque elit. Etiam et feugiat lorem. Maecenas tempus, eros vel cursus commodo, neque erat placerat nisl, nec scelerisque ipsum urna ac ante. In purus odio, feugiat quis vestibulum at, tempor nec neque. Quisque laoreet sapien in ex malesuada porttitor eu ut nisl. Etiam hendrerit elementum bibendum. Aliquam pharetra feugiat porttitor. Aliquam convallis dolor eu porta pellentesque. Proin ultricies tempor ex, ac convallis est ornare et. Donec sodales, ex nec sodales rhoncus, nunc metus facilisis massa, sit amet dapibus eros neque eget dolor. Nunc dignissim nulla ex, et luctus ante semper sit amet. Aenean id felis porta, facilisis arcu id, malesuada ex. Aenean scelerisque magna vitae posuere consectetur.

Curabitur at euismod nunc. Praesent sagittis lorem quis rutrum ornare. Vivamus ut viverra ligula. Aenean porta, lectus non pretium aliquam, leo ante varius quam, a sodales lectus urna ut tellus. In id nulla libero. Maecenas ut tempor justo. Integer diam erat, consectetur tempor dictum sed, lacinia ac elit. Vestibulum porta nulla odio, in vestibulum quam volutpat quis.

Nunc pulvinar odio in sapien molestie, quis elementum dolor porta. Nunc dapibus rutrum eros, ac varius elit rutrum ac. Cras et facilisis nunc, et consequat risus. Vivamus id maximus quam, et lacinia magna. In aliquet auctor sem. Duis finibus nisi ligula, vel tempor magna vehicula at. In neque tellus, ornare quis augue sit amet, luctus malesuada metus.


<!--END_SECTION:file:sample.md-->

## End
"
`;

exports[`@pseger/markdown-fences #injectJsDoc replace the tags 1`] = `
"# Start

<!--START_SECTION:jsdoc-->
## API

<!-- Generated by documentation.js. Update this documentation by updating the source code. -->

### foo

#### Properties

*   \`namnam\` **[String][1]** nom nom
*   \`keys\` **[Array][2]<[String][1]>** the keys

#### foo

foo test

##### Parameters

*   \`st\` **[String][1]** something
*   \`end\` **[boolean][3]** more of something

Returns **[Promise][4]<[boolean][3]>** 

#### boo

boo test

##### Parameters

*   \`st\` **[String][1]?** something
*   \`end\` **[boolean][3]** more something (optional, default \`false\`)

Returns **boom** 

### foo

Foo test

Type: {boo: exports.boo, foo: exports.foo}

#### foo

foo test

##### Parameters

*   \`st\` **[String][1]** something
*   \`end\` **[boolean][3]** more of something

Returns **[Promise][4]<[boolean][3]>** 

#### boo

boo test

##### Parameters

*   \`st\` **[String][1]?** something
*   \`end\` **[boolean][3]** more something (optional, default \`false\`)

Returns **boom** 

[1]: https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/String

[2]: https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/Array

[3]: https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/Boolean

[4]: https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/Promise

<!--END_SECTION:jsdoc-->

## End"
`;

exports[`@pseger/markdown-fences #injectJsDoc should skip the block 1`] = `
"# Start

<!--START_SECTION:deep1/deep2/insert.md-->
<!--END_SECTION:deep1/deep2/insert.md-->

## End"
`;

exports[`@pseger/markdown-fences #readContents should read the file sample.md from the test harness directory 1`] = `
"# Sample File

## First Heading

Lorem ipsum dolor sit amet, consectetur adipiscing elit. Pellentesque metus massa, ultrices nec nulla in, commodo feugiat quam. Etiam semper semper nisi at ultrices. Vestibulum blandit enim vitae ex consectetur, eu dignissim lorem vulputate. In sit amet dictum metus, in sollicitudin mi. Phasellus finibus velit sapien, at fringilla odio vehicula eget. Proin venenatis bibendum dictum. Ut iaculis nulla ac neque molestie, eget semper lorem molestie. Maecenas interdum arcu eu diam vehicula elementum. Etiam iaculis erat at nulla fringilla, non ullamcorper libero lobortis. Suspendisse potenti. Ut vitae elementum diam. Aliquam nec mauris iaculis, placerat ligula ut, dapibus justo. Nulla sodales urna vel est eleifend interdum. Sed vulputate, justo sit amet facilisis tristique, dolor tellus tincidunt nibh, sit amet aliquet turpis risus quis est.

Sed ut ante malesuada, varius velit dignissim, suscipit neque. Nullam id nulla a velit iaculis lobortis. Nam finibus diam libero, eget blandit augue lacinia at. Etiam cursus ut nibh vel dignissim. Praesent consectetur in enim eu vestibulum. Ut vestibulum pellentesque quam non pulvinar. Mauris hendrerit molestie lacinia. Sed elementum tortor suscipit, pharetra nisl at, pellentesque purus.

Fusce sagittis sem sit amet nunc condimentum pharetra. Cras in enim commodo, pretium neque a, scelerisque elit. Etiam et feugiat lorem. Maecenas tempus, eros vel cursus commodo, neque erat placerat nisl, nec scelerisque ipsum urna ac ante. In purus odio, feugiat quis vestibulum at, tempor nec neque. Quisque laoreet sapien in ex malesuada porttitor eu ut nisl. Etiam hendrerit elementum bibendum. Aliquam pharetra feugiat porttitor. Aliquam convallis dolor eu porta pellentesque. Proin ultricies tempor ex, ac convallis est ornare et. Donec sodales, ex nec sodales rhoncus, nunc metus facilisis massa, sit amet dapibus eros neque eget dolor. Nunc dignissim nulla ex, et luctus ante semper sit amet. Aenean id felis porta, facilisis arcu id, malesuada ex. Aenean scelerisque magna vitae posuere consectetur.

Curabitur at euismod nunc. Praesent sagittis lorem quis rutrum ornare. Vivamus ut viverra ligula. Aenean porta, lectus non pretium aliquam, leo ante varius quam, a sodales lectus urna ut tellus. In id nulla libero. Maecenas ut tempor justo. Integer diam erat, consectetur tempor dictum sed, lacinia ac elit. Vestibulum porta nulla odio, in vestibulum quam volutpat quis.

Nunc pulvinar odio in sapien molestie, quis elementum dolor porta. Nunc dapibus rutrum eros, ac varius elit rutrum ac. Cras et facilisis nunc, et consequat risus. Vivamus id maximus quam, et lacinia magna. In aliquet auctor sem. Duis finibus nisi ligula, vel tempor magna vehicula at. In neque tellus, ornare quis augue sit amet, luctus malesuada metus.

## Second Heading

<!--START_SECTION:jsdoc-->
<!--END_SECTION:jsdoc-->

## Third Heading

Lorem ipsum dolor sit amet, consectetur adipiscing elit. Pellentesque metus massa, ultrices nec nulla in, commodo feugiat quam. Etiam semper semper nisi at ultrices. Vestibulum blandit enim vitae ex consectetur, eu dignissim lorem vulputate. In sit amet dictum metus, in sollicitudin mi. Phasellus finibus velit sapien, at fringilla odio vehicula eget. Proin venenatis bibendum dictum. Ut iaculis nulla ac neque molestie, eget semper lorem molestie. Maecenas interdum arcu eu diam vehicula elementum. Etiam iaculis erat at nulla fringilla, non ullamcorper libero lobortis. Suspendisse potenti. Ut vitae elementum diam. Aliquam nec mauris iaculis, placerat ligula ut, dapibus justo. Nulla sodales urna vel est eleifend interdum. Sed vulputate, justo sit amet facilisis tristique, dolor tellus tincidunt nibh, sit amet aliquet turpis risus quis est.

Sed ut ante malesuada, varius velit dignissim, suscipit neque. Nullam id nulla a velit iaculis lobortis. Nam finibus diam libero, eget blandit augue lacinia at. Etiam cursus ut nibh vel dignissim. Praesent consectetur in enim eu vestibulum. Ut vestibulum pellentesque quam non pulvinar. Mauris hendrerit molestie lacinia. Sed elementum tortor suscipit, pharetra nisl at, pellentesque purus.

Fusce sagittis sem sit amet nunc condimentum pharetra. Cras in enim commodo, pretium neque a, scelerisque elit. Etiam et feugiat lorem. Maecenas tempus, eros vel cursus commodo, neque erat placerat nisl, nec scelerisque ipsum urna ac ante. In purus odio, feugiat quis vestibulum at, tempor nec neque. Quisque laoreet sapien in ex malesuada porttitor eu ut nisl. Etiam hendrerit elementum bibendum. Aliquam pharetra feugiat porttitor. Aliquam convallis dolor eu porta pellentesque. Proin ultricies tempor ex, ac convallis est ornare et. Donec sodales, ex nec sodales rhoncus, nunc metus facilisis massa, sit amet dapibus eros neque eget dolor. Nunc dignissim nulla ex, et luctus ante semper sit amet. Aenean id felis porta, facilisis arcu id, malesuada ex. Aenean scelerisque magna vitae posuere consectetur.

Curabitur at euismod nunc. Praesent sagittis lorem quis rutrum ornare. Vivamus ut viverra ligula. Aenean porta, lectus non pretium aliquam, leo ante varius quam, a sodales lectus urna ut tellus. In id nulla libero. Maecenas ut tempor justo. Integer diam erat, consectetur tempor dictum sed, lacinia ac elit. Vestibulum porta nulla odio, in vestibulum quam volutpat quis.

Nunc pulvinar odio in sapien molestie, quis elementum dolor porta. Nunc dapibus rutrum eros, ac varius elit rutrum ac. Cras et facilisis nunc, et consequat risus. Vivamus id maximus quam, et lacinia magna. In aliquet auctor sem. Duis finibus nisi ligula, vel tempor magna vehicula at. In neque tellus, ornare quis augue sit amet, luctus malesuada metus.


## Fourth Heading

<!--START_SECTION:deep1/deep2/insert.md-->
<!--END_SECTION:deep1/deep2/insert.md-->

## Third Heading

Lorem ipsum dolor sit amet, consectetur adipiscing elit. Pellentesque metus massa, ultrices nec nulla in, commodo feugiat quam. Etiam semper semper nisi at ultrices. Vestibulum blandit enim vitae ex consectetur, eu dignissim lorem vulputate. In sit amet dictum metus, in sollicitudin mi. Phasellus finibus velit sapien, at fringilla odio vehicula eget. Proin venenatis bibendum dictum. Ut iaculis nulla ac neque molestie, eget semper lorem molestie. Maecenas interdum arcu eu diam vehicula elementum. Etiam iaculis erat at nulla fringilla, non ullamcorper libero lobortis. Suspendisse potenti. Ut vitae elementum diam. Aliquam nec mauris iaculis, placerat ligula ut, dapibus justo. Nulla sodales urna vel est eleifend interdum. Sed vulputate, justo sit amet facilisis tristique, dolor tellus tincidunt nibh, sit amet aliquet turpis risus quis est.

Sed ut ante malesuada, varius velit dignissim, suscipit neque. Nullam id nulla a velit iaculis lobortis. Nam finibus diam libero, eget blandit augue lacinia at. Etiam cursus ut nibh vel dignissim. Praesent consectetur in enim eu vestibulum. Ut vestibulum pellentesque quam non pulvinar. Mauris hendrerit molestie lacinia. Sed elementum tortor suscipit, pharetra nisl at, pellentesque purus.

Fusce sagittis sem sit amet nunc condimentum pharetra. Cras in enim commodo, pretium neque a, scelerisque elit. Etiam et feugiat lorem. Maecenas tempus, eros vel cursus commodo, neque erat placerat nisl, nec scelerisque ipsum urna ac ante. In purus odio, feugiat quis vestibulum at, tempor nec neque. Quisque laoreet sapien in ex malesuada porttitor eu ut nisl. Etiam hendrerit elementum bibendum. Aliquam pharetra feugiat porttitor. Aliquam convallis dolor eu porta pellentesque. Proin ultricies tempor ex, ac convallis est ornare et. Donec sodales, ex nec sodales rhoncus, nunc metus facilisis massa, sit amet dapibus eros neque eget dolor. Nunc dignissim nulla ex, et luctus ante semper sit amet. Aenean id felis porta, facilisis arcu id, malesuada ex. Aenean scelerisque magna vitae posuere consectetur.

Curabitur at euismod nunc. Praesent sagittis lorem quis rutrum ornare. Vivamus ut viverra ligula. Aenean porta, lectus non pretium aliquam, leo ante varius quam, a sodales lectus urna ut tellus. In id nulla libero. Maecenas ut tempor justo. Integer diam erat, consectetur tempor dictum sed, lacinia ac elit. Vestibulum porta nulla odio, in vestibulum quam volutpat quis.

Nunc pulvinar odio in sapien molestie, quis elementum dolor porta. Nunc dapibus rutrum eros, ac varius elit rutrum ac. Cras et facilisis nunc, et consequat risus. Vivamus id maximus quam, et lacinia magna. In aliquet auctor sem. Duis finibus nisi ligula, vel tempor magna vehicula at. In neque tellus, ornare quis augue sit amet, luctus malesuada metus.

"
`;

exports[`@pseger/markdown-fences generateMarkDownFile full integration test 1`] = `
"# Test

<!--START_SECTION:jsdoc-->
## API

<!-- Generated by documentation.js. Update this documentation by updating the source code. -->

### foo

#### Properties

*   \`namnam\` **[String][1]** nom nom
*   \`keys\` **[Array][2]<[String][1]>** the keys

#### foo

foo test

##### Parameters

*   \`st\` **[String][1]** something
*   \`end\` **[boolean][3]** more of something

Returns **[Promise][4]<[boolean][3]>** 

#### boo

boo test

##### Parameters

*   \`st\` **[String][1]?** something
*   \`end\` **[boolean][3]** more something (optional, default \`false\`)

Returns **boom** 

### foo

Foo test

Type: {boo: exports.boo, foo: exports.foo}

#### foo

foo test

##### Parameters

*   \`st\` **[String][1]** something
*   \`end\` **[boolean][3]** more of something

Returns **[Promise][4]<[boolean][3]>** 

#### boo

boo test

##### Parameters

*   \`st\` **[String][1]?** something
*   \`end\` **[boolean][3]** more something (optional, default \`false\`)

Returns **boom** 

[1]: https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/String

[2]: https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/Array

[3]: https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/Boolean

[4]: https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/Promise

<!--END_SECTION:jsdoc-->

<!--START_SECTION:file:INSTALL.md-->
## Install

\`\`\`javascript
const hello = 'hi';
\`\`\`

<!--END_SECTION:file:INSTALL.md-->

--fini--
"
`;

exports[`@pseger/markdown-fences injectToc replace the tags 1`] = `
"# Start

<!--START_SECTION:toc-->

## Table of contents
- [Start](#start)
  * [Level 2-A](#level-2-a)
    + [Level 3-A](#level-3-a)
    + [Level 3-B](#level-3-b)
    + [Level 3-C](#level-3-c)
  * [Level 2-B](#level-2-b)
    + [Level 3-A](#level-3-a-1)
      - [Level 4-A](#level-4-a)
      - [Level 4-B](#level-4-b)
    + [Level 3-C](#level-3-c-1)
  * [Level 2-C](#level-2-c)

<!--END_SECTION:toc-->

## Level 2-A

### Level 3-A

### Level 3-B

### Level 3-C

## Level 2-B

### Level 3-A

#### Level 4-A

#### Level 4-B

### Level 3-C

## Level 2-C

"
`;

exports[`@pseger/markdown-fences injectToc should skip the block 1`] = `
"# Start

<!--START_SECTION:blaablaa-->
<!--END_SECTION:blaablaa-->

## Level 2-A

### Level 3-A

### Level 3-B

### Level 3-C

## Level 2-B

### Level 3-A

#### Level 4-A

#### Level 4-B

### Level 3-C

## Level 2-C

"
`;
